bridgev2/networkinterface.go

```diff
@@ -593,6 +593,44 @@ type GroupCreatingNetworkAPI interface {
 	CreateGroup(ctx context.Context, name string, users ...networkid.UserID) (*CreateChatResponse, error)
 }
 
+type MembershipChangeType struct {
+	From   event.Membership
+	To     event.Membership
+	IsSelf bool
+}
+
+var (
+	AcceptInvite  = MembershipChangeType{From: event.MembershipInvite, To: event.MembershipJoin, IsSelf: true}
+	RevokeInvite  = MembershipChangeType{From: event.MembershipInvite, To: event.MembershipLeave}
+	RejectInvite  = MembershipChangeType{From: event.MembershipInvite, To: event.MembershipLeave, IsSelf: true}
+	BanInvited    = MembershipChangeType{From: event.MembershipInvite, To: event.MembershipBan}
+	ProfileChange = MembershipChangeType{From: event.MembershipJoin, To: event.MembershipJoin, IsSelf: true}
+	Leave         = MembershipChangeType{From: event.MembershipJoin, To: event.MembershipLeave, IsSelf: true}
+	Kick          = MembershipChangeType{From: event.MembershipJoin, To: event.MembershipLeave}
+	BanJoined     = MembershipChangeType{From: event.MembershipJoin, To: event.MembershipBan}
+	Invite        = MembershipChangeType{From: event.MembershipLeave, To: event.MembershipInvite}
+	Join          = MembershipChangeType{From: event.MembershipLeave, To: event.MembershipJoin}
+	Ban           = MembershipChangeType{From: event.MembershipLeave, To: event.MembershipBan}
```

Comment:
This should have the from membership to be consistent with the others (it's not even the most common case, so having a generic name doesn't make sense)

```suggestion
	BanLeft       = MembershipChangeType{From: event.MembershipLeave, To: event.MembershipBan}
```